### PR TITLE
define new keyboard "angel64"

### PR DESCRIPTION
**Pull request by kakunpc**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [X] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/angel64/rules.mk [hunk 1]
```diff
@@ -0,0 +1,84 @@
+# MCU name
+#MCU = at90usb1286
```
Comment (fauxpark) [inline keyboards/angel64/rules.mk]:
```suggestion
```

**Comment by drashna** [comment]
Is there a reason for the custom matrix here?  

It looks like all it's doing is ... well running both the col2row and row2col scans, which .... sounds like it's not needed.

**Comment by kakunpc** [comment]
Both COL2ROW and ROW2COL must be performed.
Can I define this pattern without using custom matrix?

SnapCrab_NoName_2019-7-13_10-53-59_No-00

**Comment by kakunpc** [comment]
This is the matrix that requires col2row and row2col.

SnapCrab_NoName_2019-7-13_10-53-59_No-00

**Comment by drashna** [comment]
Okay.  

At some point, it may be worth importing the matrix stuff into the core code, with something like `COL2ROW2COL` and/or `ROW2COL2ROW`, or some such. 

But for now, this look good.  Thanks!

**Pull request merged**